### PR TITLE
tools: Added yml version of tools-version files

### PR DESCRIPTION
**Pull request by DatGizmo**
Jira: NCSDK-20076

### CODE REVIEW

File: scripts/tools-versions-win10.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+python:
```
Comment (gmarull) [inline scripts/tools-versions-win10.yml:1]: note: I see a lot of duplication here, should we have a common file, that is imported by OS specific files where versions can be overriden if needed?

**Review by gmarull — COMMENTED** [review]
These changes mean that we need to find an alternative to https://github.com/nrfconnect/sdk-nrf/blob/main/doc/update_versions.cmake and invoking code here https://github.com/nrfconnect/sdk-nrf/blob/main/doc/CMakeLists.txt#L210-L245. Which end up populating https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/versions.txt.in. Maybe it's time to create a Sphinx extension, so that we can place in the docs something like:

```
:tool-version:`cmake`
```

cc: @carlescufi @gautesl

File: scripts/tools-versions-win10.yml (hunk 1 above)
Comment (jangalda-nsc) [reply scripts/tools-versions-win10.yml:1]: It is not possible at this moment.
To do so, changes in toolchain bundler are required. Only platform specific files are evaluated during bundle preparation process. They are also used to calculate bundle checksum, so without changes in bundler we cannot extract common requirements to another file

**Comment by gautesl** [comment]
A new Sphinx extension mechanism for scanning tool versions were introduced in #10929, which can also parse yml files. This can be set by changing the filenames to be parsed in https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/conf.py#L182-L186 from .txt to the newly introduced .yml files.

File: scripts/tools-versions-linux.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    version: 1.4.7-3ubuntu2
+west:
+    version: 1.0.0
+gn:
+    version: e55664a01f6ee92762210d0c01b13552f8117e11
+nanopb:
+    version: 0.4.6
+zephyr-sdk:
+    version: 0.16.0
+    architectures: # https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.15.2
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply scripts/tools-versions-linux.yml:21]: The RISCV-64 compiler is not strictly required, but there's no harm in including it.

[the remark this answers is not in the document]
Comment (DatGizmo) [reply scripts/tools-versions-linux.yml]: Fixed url, and removed the riscv compiler.

**Review by gmarull — CHANGES_REQUESTED** [review]
please update doc/nrf/conf.py so that these files are used.

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**